sdk/core/loader/types.h

```diff
@@ -482,6 +482,15 @@ namespace loader
 			{
 				return code.start();
 			}
+
+			/**
+			 * Privileged libraries are privileged compartments without data
+			 * segments.
+			 */
+			[[nodiscard]] bool is_privileged_library() const
+			{
+				return (data.start() == 0) && (data.size() == 0);
```

Comment:
This possibly should check that the length is 0.  It's possible that some memory layout will make address 0 usable memory.